r10.go

```diff
@@ -0,0 +1,239 @@
+// r10: all the previous optimizations plus faster semicolon finding and
+// hashing
+//
+// 24.8x faster than r1
+// 47.62ms for 10M rows on Menno's laptop (which is slower than Ben's)
```

Comment:
Would you mind also leaving a link to https://github.com/gunnarmorling/1brc/blob/main/src/main/java/dev/morling/onebrc/CalculateAverage_mtopolnik.java in a code comment?

Reply:
Thanks for taking care of this. We've had guests staying with us so I haven't had a chance to get back to this.